Client/Extensions/PaymentRequestExtension.swift

```diff
@@ -0,0 +1,142 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import BraveRewards
+import BraveShared
+import Data
+import Shared
+import WebKit
+
+private let log = Logger.browserLogger
+
+let popup = PaymentHandlerPopupView(imageView: nil, title: Strings.paymentRequestTitle, message: "")
+
+class PaymentRequestExtension: NSObject {
+    fileprivate weak var tab: Tab?
+    fileprivate weak var rewards: BraveRewards?
+    fileprivate var response = ""
+    fileprivate var token: String
+    
+    fileprivate enum PaymentRequestErrors: String {
+        case NotSupportedError = "NotSupportedError"
+        case AbortError = "AbortError"
+    }
+    
+    init(rewards: BraveRewards, tab: Tab) {
+        token = UserScriptManager.securityToken.uuidString.replacingOccurrences(of: "-", with: "", options: .literal)
+        self.tab = tab
+        self.rewards = rewards
+    }
+}
+
+extension PaymentRequestExtension: TabContentScript {
+    static func name() -> String {
+        return "PaymentRequest"
+    }
+    
+    func scriptMessageHandlerName() -> String? {
+        return PaymentRequestExtension.name()
+    }
+    
+    func sendPaymentRequestError(errorName: String, errorMessage: String) {
+        ensureMainThread {
+            self.tab?.webView?.evaluateJavaScript("PaymentRequestCallback\(self.token).paymentreq_postCreate('', '\(errorName)', '\(errorMessage)')") { _, error in
```

Comment:
does this need any sanitization against XSS type attacks?

Reply:
`errorName` and `errorMessage` are constant strings on the client side. 

1. https://github.com/brave/brave-ios/pull/2327/files/6b86149b52450d9fcb4ef76b79aea9decb58bc78#diff-4dc70f536f408cb9c86ed24f1c0394eeR21

2. https://github.com/brave/brave-ios/pull/2327/files/6b86149b52450d9fcb4ef76b79aea9decb58bc78#diff-4dc70f536f408cb9c86ed24f1c0394eeR139